### PR TITLE
Cd 49 admin global configuration dropdown options backend

### PR DESCRIPTION
**Pull request by BansariBhatt**

### CODE REVIEW

File: CIR.Application/Services/GlobalConfiguration/GlobalConfigurationReasonsService.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using CIR.Core.Entities.GlobalConfiguration;
+using CIR.Core.Interfaces.GlobalConfiguration;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CIR.Application.Services.GlobalConfiguration
+{
+    public class GlobalConfigurationReasonsService : IGlobalConfigurationReasonsService
+    {
+        private readonly IGlobalConfigurationReasonsRepository _globalConfigurationReasonsRepository;
```
Comment (GohilAmit) [inline CIR.Application/Services/GlobalConfiguration/GlobalConfigurationReasonsService.cs:9]: not use underscore

File: CIR.Data/Data/GlobalConfiguration/GlobalConfigurationReasonsRepository.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+using Microsoft.AspNetCore.Mvc;
+using System.Data;
+
+namespace CIR.Data.Data.GlobalConfiguration
+{
+    public class GlobalConfigurationReasonsRepository : IGlobalConfigurationReasonsRepository
+    {
+        #region PROPERTIES
+
+        private readonly CIRDbContext _CIRDbContext;
```
Comment (GohilAmit) [inline CIR.Data/Data/GlobalConfiguration/GlobalConfigurationReasonsRepository.cs:16]: object name define camelCase

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+namespace CIR.Controllers.GlobalConfiguration
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    //[Authorize]
+    public class GlobalConfigurationReasonsController : Controller
+    {
+        #region PROPERTIES
+
+        private readonly IGlobalConfigurationReasonsService _globalConfigurationReasonsService;
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs:17]: not use underscore

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    return await _globalConfigurationReasonsService.CreateOrUpdateGlobalConfigurationReasons(globalConfigurationReasons);
+                }
+                catch (Exception ex)
+                {
+                    return new JsonResult(new CustomResponse<Exception>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.InternalServerError, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.InternalServerError.GetDescriptionAttribute(), Data = ex });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs:66]: need to change catch return response

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                }
+            }
+            return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.BadRequest, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.BadRequest.GetDescriptionAttribute(), Data = SystemMessages.msgBadRequest });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs:69]: need to change return response message

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    return await _globalConfigurationReasonsService.CreateOrUpdateGlobalConfigurationReasons(globalConfigurationReasons);
+                }
+                catch (Exception ex)
+                {
+                    return new JsonResult(new CustomResponse<Exception>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.InternalServerError, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.InternalServerError.GetDescriptionAttribute(), Data = ex });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs:89]: need to change catch return response

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+                }
+            }
+            return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.BadRequest, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.BadRequest.GetDescriptionAttribute(), Data = SystemMessages.msgBadRequest });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationReasonsController.cs:92]: need to change  return response message

**Review by GohilAmit — APPROVED** [review]

**Review by mittalvyas1 — APPROVED** [review]

**Review by jayk-patel-tatvasoft — APPROVED** [review]

**Pull request merged**